### PR TITLE
Feature/white238/loosen conduit variants

### PR DESCRIPTION
**Pull request by white238**
# Summary

- This PR is a bugfix/feature
- It does the following :
  - Removes Axom's Conduit variant allowing users to override it
  - Fixes Conduit's shared library rpath'ing on rzansel w/ clang/xlf
  - Skips chmod/chgrp'ing on non-shared locations
  - Move devtools to shared /collab location so non-axom group members can use them

I'll push these changes to spack as well

### CODE REVIEW

File: scripts/llnl_scripts/llnl_lc_build_tools.py [hunk 1]
```diff
@@ -521,18 +521,30 @@ def set_axom_group_and_perms(directory):
...
+
+    skip = True
+    shared_dirs = [get_shared_base_dir(), get_shared_collab_dir(), get_archive_base_dir()]
+    for shared_dir in shared_dirs:
+        if directory.startswith(shared_dir):
+            skip = False
+            break
+
+    if skip:
+        print "[Skipping, due to not known shared location, changing group and access perms of: %s]" % directory
```
Comment (kennyweiss) [inline scripts/llnl_scripts/llnl_lc_build_tools.py:533]: This message can probably be clarified. Maybe something like:
> Skipping update of group and access permissions. Provided directory '{directory}' not was outside shared locations

**Comment by rhornung67** [comment]
> Thank you for your service, @white238 .

Indeed!

**Comment by white238** [comment]
Needless to say this has gone down a rabbit hole.  Loosening the MPI variant on MFEM has gone very wrong. This is needed for two reasons, a user complained through the Spack channels and also to enable @joshessman-llnl 's sidre datastore class moving.

I have ideas on how to "fix" this but all my ideas are kind of gross.

1) add a "parmfem" variant that is on by default but will turn off MFEM's mpi variant
2) debug why hypre fails on so many of our compiler specs
3) Greg Becker is currently looking into a cleaner way to turn off mfem variants this per spec (temporary solution though)

At this point I am just trying to get back to passing all specs even if I have to turn off MFEM on some of them.

**Comment by white238** [comment]
Closing because the important parts have been done in newer PRs

**Pull request closed without merging**